pkg/serverless/aws/arn.go

```diff
@@ -115,3 +160,89 @@ func RestoreCurrentStateFromFile() error {
 	SetRequestID(restoredState.CurrentReqID)
 	return nil
 }
+
+// BuildFunctionARNFromEnv reconstructs the function arn from what's available
+// in the environment.
+func BuildFunctionARNFromEnv() string {
```

Comment:
Nit: I would rename this function to `GetFunctionARN()`. The fact that it does so by building it from the environment (and also making a call to STS) is an implementation detail and not important to what the function actually does.

Reply:
Also, I might add some tests to cover the edge cases in this function.

Reply:
Oh, I think you were probably trying to differentiate this from `GetARN()`. Never mind then.

Reply:
Yeah, I wanted to indicate it's doing work that might take a bit of time, and shouldn't be called frequently.